### PR TITLE
Move stable REST API migration guide to UPDATING.md

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This PR moves Stable REST API migration guide to UPDATING.md. This will enable us to have a common place where all updating information is available.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
the old doc is still referenced and fails docs build

File: UPDATING.md [hunk 1]
```diff
@@ -69,6 +69,47 @@ More tips can be found in the guide:
 https://developers.google.com/style/inclusive-documentation
 
 -->
+### Migration Guide from Experimental API to Stable API v1
+If your application is still using the experimental API, it is important to
+consider migrating to the stable API so that your application continues to
+work.
```
Comment (mik-laj) [inline UPDATING.md:75]:
```suggestion
work. Experimental API still works in Airflow 2.0, but support may be dropped in the future.
```

File: UPDATING.md [hunk 2]
```diff
@@ -69,6 +69,47 @@ More tips can be found in the guide:
...
+
+The stable API exposes many endpoints available through the webserver. Here are the
+differences between the two endpoints that will help you migrate from the
+experimental REST API to the stable REST API.
```
Comment (mik-laj) [inline UPDATING.md]:
```suggestion

In Airflow 2.0, we added the new REST API. Experimental API still works, but support may be dropped in the future. If your application is still using the experimental API, you should
consider migrating to the stable API.

The stable API exposes many endpoints available through the webserver. Here are the
differences between the two endpoints that will help you migrate from the
experimental REST API to the stable REST API.
```

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Needs rebase now :(

**Review by mik-laj — APPROVED** [review]

**Review by turbaszek — APPROVED** [review]

**Review by mik-laj — APPROVED** [review]

**Pull request merged**